secret/secrets_manager_client_test.go

```diff
@@ -61,11 +72,77 @@ func TestSecretsManagerCreateAndDeleteSecret(t *testing.T) {
 		require.NoError(t, err)
 		require.NotZero(t, out)
 
+		defer Cleanup(out)
+
+		require.NotZero(t, out.Name)
+		require.NotZero(t, out.ARN)
+
+	})
+
+	t.Run("GetFailsWithInvalidInput", func(t *testing.T) {
+		out, err := c.GetSecretValue(ctx, &secretsmanager.GetSecretValueInput{})
+		assert.Error(t, err)
+		assert.Zero(t, out)
+	})
+
+	t.Run("UpdateFailsWithInvalidInput", func(t *testing.T) {
+		out, err := c.UpdateSecret(ctx, &secretsmanager.UpdateSecretInput{})
+		assert.Error(t, err)
+		assert.Zero(t, out)
+	})
+
+	t.Run("CreateAndGetSucceed", func(t *testing.T) {
+		out, err := c.CreateSecret(ctx, &secretsmanager.CreateSecretInput{
+			Name:         aws.String(os.Getenv("AWS_SECRET_PREFIX") + "foobar"),
+			SecretString: aws.String("barfoo"),
+		})
+		require.NoError(t, err)
+		require.NotZero(t, out)
+
+		defer Cleanup(out)
+
+		defer func() {
+			if out != nil && out.Name != nil && out.ARN != nil {
+				out, err := c.GetSecretValue(ctx, &secretsmanager.GetSecretValueInput{
+					SecretId: out.ARN,
+				})
+				require.NoError(t, err)
+				require.NotZero(t, out)
+				assert.Equal(t, "barfoo", *out.SecretString)
+				assert.Equal(t, os.Getenv("AWS_SECRET_PREFIX")+"foobar", *out.Name)
+			}
+		}()
+
+		require.NotZero(t, out.Name)
+		require.NotZero(t, out.ARN)
+	})
+
+	t.Run("CreateAndUpdateAndGetSucceed", func(t *testing.T) {
+		out, err := c.CreateSecret(ctx, &secretsmanager.CreateSecretInput{
+			Name:         aws.String(os.Getenv("AWS_SECRET_PREFIX") + "mongodb"),
```

Comment:
Rather than keep copy-pasting `os.Getenv("AWS_SECRET_PREFIX")` and appending a hard-coded string can you make a helper function to do this? For example:

```go
func makeTestSecret(name string) {
    return fmt.Sprintf(path.Join(os.Getenv("AWS_SECRET_PREFIX"), "cocoa", name)
}
```

I would include the project name (cocoa) so that we know which project the secret came from. I would also use the test name in the secret name so that it's obvious where a secret came from if it fails to clean up. Lastly, I would include just a random string afterwards for uniqueness to ensure that secrets don't collide (which you can get from `utility.RandomString()`) - if two Evergreen tasks in different patches run at the same time, they might conflict and tests should always be isolated from one another.

For example, instead of this, I would write:
```go
aws.String(makeTestSecret(t.Name() + utility.RandomString())
```
Which would generate a string like `${AWS_SECRET_PREFIX}/cocoa/CreateAndUpdateAndGetSucceed<RANDOM_STRING>`.

Reply:
Can I add the `utility.RandomString()` at the end as a parameter of `path.Join`, or will I have to append it as an input after every call to `utility.RandomString()`

Reply:
You can add it to the function itself if you want.

The only caveat with a randomized string is that if you want to check that the secret name you pass in matches in the output, you'll have to store `makeTestSecret(<stuff>)` in a variable.